### DISCUSSION TITLE
How does RCVD_VIA_SMTP_AUTH work?

### DISCUSSION THREAD

**Discussion by reox** [rspamd/rspamd#4744] [Q&A] [answered]
I'm testing a relay mail server, by sending mails over it to my account which is behind a rspamd. However, I noticed that the rule RCVD_UNAUTH_PBL triggers, because RECEIVED_SPAMHAUS_PBL is present and apparently RCVD_VIA_SMTP_AUTH is not. It is no wonder that the spamhaus rule triggerd, because I send the mail from a dynamic v4 address - possibly used earlier by another person for spamming.
However, what I do not understand is how RCVD_VIA_SMTP_AUTH  is not present. The relay I use makes use of authentication and is definitely not an open relay.

I checked the code and I think this is where this flag is evaluated: https://github.com/rspamd/rspamd/blob/b0e98a56befeed42291de8ac01d46f86879c6c97/rules/misc.lua#L241

Am I right, that it might be a problem if the first Received header is not by a "localhost" but by some other host?
Because for the mail server that does not work, the header looks like this:
```
Received: from [127.0.0.1] (x-y-z-a.dynamic-adsl.xxxxxx [x.y.z.a])
	(authenticated bits=0)
	by mailserver.example.com (...) with ESMTP id 12345
	(version=TLSv1/SSLv3 cipher=AES128-GCM-SHA256 bits=128 verify=NO)
	for <someone@example.com>; Thu, 14 Dec 2023 11:49:41 +0100
```

Thus the evaluation fails, as there is never any host named "localhost" in the received chain and thus the rule thinks that it was send unauthenticated?

**Comment by fatalbanana** [comment, marked as answer]
It's expecting to see protocol like `ESMTPA` or `ESMTPSA`.

**Reply by reox** [reply]
Ah okay, I see! I.e., https://www.rfc-editor.org/rfc/rfc3848.html right? I did not knew that one.
So in this case it should indeed write ESMTPSA - I'll check why it does not. Thanks!

**Reply by chkpnt** [reply]
This leads to a false positive when the sender used a webmailer like t-online.de or web.de as `ESMTPA` or `ESMTPSA` is missing:

t-online.de:

<pre>
Received: from mailout12.t-online.de (mailout12.t-online.de [194.25.134.22])
	(using TLSv1.3 with cipher TLS_AES_256_GCM_SHA384 (256/256 bits)
	 key-exchange X25519 server-signature RSA-PSS (4096 bits) server-digest SHA256)
	(No client certificate requested)
	by mail.example.com (Postfix) with ESMTPS id 4DDF15A6CDB9
	for <mail@example.com>; Thu, 12 Dec 2024 21:32:01 +0100 (CET)
Received: from fwd70.aul.t-online.de (fwd70.aul.t-online.de [10.223.144.96])
	by mailout12.t-online.de (Postfix) with SMTP id 29C708F1
	for <mail@example.com>; Thu, 12 Dec 2024 21:32:01 +0100 (CET)
Received: from spica15.mgt.mul.t-online.de ([172.20.102.138]) by fwd70.aul.t-online.de
	with esmtp id 1tLpqh-2BZ6u00; Thu, 12 Dec 2024 21:31:51 +0100
Received: <b>from 109.250.137.110:50366 by cmpweb01.aul.t-online.de with HTTP/1.1</b> (Lisa V8-0-3-0.0 on API V5-54-1-0); Thu, 12 Dec 24 21:31:51 +0100
Received: from 172.20.102.128:51940 by spica15.mgt.mul.t-online.de:8080; Thu, 12 Dec 2024 21:31:51 +0100 (CET)
X-Spamd-Result: default: False [3.10 / 15.00];
	RCVD_UNAUTH_PBL(2.00)[];
	...
	RECEIVED_SPAMHAUS_PBL(0.00)[<b>109.250.137.110</b>:received];
</pre>

web.de:

<pre>
Received: from mout.web.de (mout.web.de [212.227.15.4])
	(using TLSv1.3 with cipher TLS_AES_256_GCM_SHA384 (256/256 bits)
	 key-exchange ECDHE (P-256) server-signature RSA-PSS (4096 bits) server-digest SHA256)
	(No client certificate requested)
	by mail.example.com (Postfix) with ESMTPS id D4C455D2C4DC
	for <mail@example.com>; Sat,  1 Feb 2025 22:35:14 +0100 (CET)
Received: <b>from [109.250.136.140] ([109.250.136.104]) by msvc-mesg-web009
 (via HTTP)</b>; Sat, 1 Feb 2025 22:35:14 +0100
X-Spamd-Result: default: False [3.80 / 15.00];
	RCVD_UNAUTH_PBL(2.00)[];
	RECEIVED_SPAMHAUS_PBL(0.00)[<b>109.250.136.140</b>:received];
</pre>

I've no issues with mails from Gmail, as they do not include the sender's original ip address in a received header.
